habitat/tasks/nav/nav.py

```diff
@@ -202,6 +203,72 @@ def get_observation(
         )
 
 
+@registry.register_sensor
+class ImageGoalSensor(Sensor):
+    r"""Sensor for ImageGoal observations which are used in ImageGoal Navigation.
+
+    RGBSensor needs to be one of the Simulator sensors.
+    Currently the task is using pointnav Dataset.
+    This sensor return the rgb image taken from the goal position to reach with
+    random rotation.
+
+    Args:
+        sim: reference to the simulator for calculating task observations.
+        config: config for the ImageGoal sensor.
+    """
+
+    def __init__(
+        self, *args: Any, sim: Simulator, config: Config, **kwargs: Any
+    ):
+        self._sim = sim
+        sensors = self._sim.sensor_suite.sensors
+        rgb_sensor_uuids = [
+            uuid
+            for uuid, sensor in sensors.items()
+            if isinstance(sensor, RGBSensor)
+        ]
+        if len(rgb_sensor_uuids) != 1:
+            raise ValueError(
+                f"ImageGoalNav requires one RGB sensor, {len(rgb_sensor_uuids)} detected"
+            )
+
+        self._rgb_sensor_uuid, = rgb_sensor_uuids
+        self._current_episode_id = None
+        self._current_image_goal = None
+        super().__init__(config=config)
+
+    def _get_uuid(self, *args: Any, **kwargs: Any):
+        return "imagegoal"
+
+    def _get_sensor_type(self, *args: Any, **kwargs: Any):
+        return SensorTypes.PATH
+
+    def _get_observation_space(self, *args: Any, **kwargs: Any):
+        return self._sim.sensor_suite.observation_spaces.spaces[
+            self._rgb_sensor_uuid
+        ]
+
+    def get_observation(
+        self, *args: Any, observations, episode: Episode, **kwargs: Any
+    ):
+        if episode.episode_id == self._current_episode_id:
+            return self._current_image_goal
+
+        self._current_episode_id = episode.episode_id
```

Comment:
Maybe, let's move image goal evaluation to separate internal method `_get_episode_image_goal` or `get_pointnav_episode_image_goal`. Then it will be easier to update the code when the dataset will have fixed rotation as well.